code-editor/scripts/python-package/src/sagemaker_code_editor_auto_shut_down/autostop.py

```diff
@@ -0,0 +1,75 @@
+import os
+import time
+import boto3
+import json
+import argparse
+
+def check_user_activity(workspace_dir, idle_threshold):
+    # Get the timestamp of the most recently modified file or folder
+    recent_item = max(
+        (os.path.join(root, file) for root, _, files in os.walk(workspace_dir) for file in files),
+        key=lambda x: os.stat(x).st_mtime,
+        default=None
+    )
+
+    # Get the current time
+    current_time = time.time()
+
+    # Calculate the time difference
+    time_diff = current_time - os.stat(recent_item).st_mtime if recent_item else float('inf')
+
+    # Check if the user is idle based on the idle time threshold
+    if time_diff > idle_threshold:
+        return "idle"
+    else:
+        return "active"
+
+# Create an argument parser
+parser = argparse.ArgumentParser(description='Check user activity and terminate SageMaker Studio app if idle.')
+parser.add_argument('--time', type=int, help='Idle time threshold in seconds')
+parser.add_argument('--region', type=str, help='AWS region')
+
+# Parse the command-line arguments
+args = parser.parse_args()
+
+# Check if idle_threshold is provided
+if args.time is None:
+    parser.print_help()
+    sys.exit(1)
+
+if args.region is None:
+    parser.print_help()
+    sys.exit(1)
+
+# Define the folders to monitor for modifications
+workspace_dirs = ["/opt/amazon/sagemaker/sagemaker-code-editor-server-data/data/User/History", "/opt/amazon/sagemaker/sagemaker-code-editor-server-data/data/User/Backups/empty-window/untitled"]
```

Comment:
It'll help to include the rationale here on why we're monitoring these two folders